cmd/contour/serve.go

```diff
@@ -200,196 +240,125 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
-		UseProxyProto: ctx.useProxyProto,
-		HTTPListeners: map[string]xdscache_v3.Listener{
-			"ingress_http": {
-				Name:    "ingress_http",
-				Address: ctx.httpAddr,
-				Port:    ctx.httpPort,
-			},
-		},
-		HTTPSListeners: map[string]xdscache_v3.Listener{
-			"ingress_https": {
-				Name:    "ingress_https",
-				Address: ctx.httpsAddr,
-				Port:    ctx.httpsPort,
-			},
-		},
-		HTTPAccessLog:                 ctx.httpAccessLog,
-		HTTPSAccessLog:                ctx.httpsAccessLog,
-		AccessLogType:                 ctx.Config.AccessLogFormat,
-		AccessLogFields:               ctx.Config.AccessLogFields,
-		AccessLogFormatString:         ctx.Config.AccessLogFormatString,
-		AccessLogFormatterExtensions:  ctx.Config.AccessLogFormatterExtensions(),
-		MinimumTLSVersion:             annotation.MinTLSVersion(ctx.Config.TLS.MinimumProtocolVersion, "1.2"),
-		CipherSuites:                  config.SanitizeCipherSuites(ctx.Config.TLS.CipherSuites),
-		RequestTimeout:                requestTimeout,
-		ConnectionIdleTimeout:         connectionIdleTimeout,
-		StreamIdleTimeout:             streamIdleTimeout,
-		DelayedCloseTimeout:           delayedCloseTimeout,
-		MaxConnectionDuration:         maxConnectionDuration,
-		ConnectionShutdownGracePeriod: connectionShutdownGracePeriod,
-		DefaultHTTPVersions:           parseDefaultHTTPVersions(ctx.Config.DefaultHTTPVersions),
-		AllowChunkedLength:            !ctx.Config.DisableAllowChunkedLength,
-		XffNumTrustedHops:             ctx.Config.Network.XffNumTrustedHops,
-		ConnectionBalancer:            ctx.Config.Listener.ConnectionBalancer,
-	}
-
-	if ctx.Config.RateLimitService.ExtensionService != "" {
-		namespacedName := k8s.NamespacedNameFrom(ctx.Config.RateLimitService.ExtensionService)
-		client := clients.DynamicClient().Resource(contour_api_v1alpha1.ExtensionServiceGVR).Namespace(namespacedName.Namespace)
-
-		// ensure the specified ExtensionService exists
-		res, err := client.Get(context.Background(), namespacedName.Name, metav1.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("error getting rate limit extension service %s: %v", namespacedName, err)
-		}
-		var extensionSvc contour_api_v1alpha1.ExtensionService
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(res.Object, &extensionSvc); err != nil {
-			return fmt.Errorf("error converting rate limit extension service %s: %v", namespacedName, err)
-		}
-		// get the response timeout from the ExtensionService
-		var responseTimeout timeout.Setting
-		if tp := extensionSvc.Spec.TimeoutPolicy; tp != nil {
-			responseTimeout, err = timeout.Parse(tp.Response)
-			if err != nil {
-				return fmt.Errorf("error parsing rate limit extension service %s response timeout: %v", namespacedName, err)
-			}
-		}
+	cipherSuites := []string{}
+	for _, cs := range contourConfiguration.Envoy.Listener.TLS.CipherSuites {
+		cipherSuites = append(cipherSuites, string(cs))
+	}
+
+	listenerConfig := xdscache_v3.NewListenerConfig(
```

Comment:
Note for later; this feels like a good candidate for just passing in the Config object or something.